plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/WhileStack.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.conversion
+
+interface WhileStackContext {
+    fun inNewWhileBlock(action: (Int) -> Unit)
+
+    fun getWhile(): Int
+}
+
+class WhileStack(
+    private val labelsIndexStack: MutableList<Int> = mutableListOf(),
+    private var nextIndex: Int = 0
+) : WhileStackContext {
+    override fun inNewWhileBlock(action: (Int) -> Unit) {
```

Comment:
I suspect having the `WhileStack` just be a single `Int` and then having `inNewWhileBlock` be in `StmtConverter` and replace the `WhileStack` will work better; so far we've been doing these context changes as immutably as possible.

Reply:
But even if `WhileStack` is a single `Int`, it should be mutable right? Because I don't know how to keep track of the current `while` ID otherwise

Reply:
Actually I am not even sure that we can do it without a stack. Maybe I am missing something

Reply:
The stack effect is achieved by replacing the `WhileStack` when you call `inNewWhileBlock`, just like how we now replace the `SeqnBuilder`.  You could give the `WhileStack` a reference to the previous `WhileStack` if that's necessary, but I don't think it will be for now; maybe later it will be needed to support breaking/continuing to a loop with a different label.

Reply:
I think that the situation is similar to the naming of anon vars in which we are currently using mutable vars.
Your solution works for nested while loops, but what about this:
```
while(false) {}
while(false) {}
```
in this case the second `while` is outside `inNewWhileBlock` of the first one, but I still want to give it a different ID

Reply:
Ah, I see.  Could you get the id from the program converter, though?  That way the `WhileStack` is immutable, but you can still use increasing indices.